In [1]:
# import all the filenames
from allfunctions import *

1  def LassoCVModel(filename):

2  def OMPCVModel(filename):

3  def GradientBoostingCVModel(filename):

4  def RandomForestCVModel(filename):

5  def RidgeCVModel(filename):

6  def ElasticNetCVModel(filename):

7  def SVRPolyCVModel(filename):

8  def SVRSigmoidCVModel(filename):

9  def SVRLinearCVModel(filename):

10 def SVRRbfCVModel(filename):

In [2]:
def wrapper():
    if __name__ == '__main__':
        global model_selection
        global functionList
        global filenames
        
        functionList = [ LassoCVModel, GradientBoostingCVModel, 
                 RandomForestCVModel, RidgeCVModel, ElasticNetCVModel, 
                 SVRPolyCVModel, SVRSigmoidCVModel, SVRLinearCVModel, 
                 SVRRbfCVModel ] #OMPCVModel

        filenames = ['AB.pickle', 'BC.pickle', 'CD.pickle', 'AC.pickle', 'AD.pickle', 'BD.pickle' ]
        
        model_selection = {}
        
        for f in functionList:
            print()
            for filename in filenames:
                out = f(filename)
                newScore = out[filename]['train_rmse_cv_4_picking']

                if filename in model_selection:
                    #compare scores and select more accurate model
                    oldScore = model_selection[filename]['train_rmse_cv_4_picking']
                    if(newScore < oldScore):
                        print("【【【】】】switching models for %s..." % filename)
                        print("【【【】】】newScore: %f < oldScore: %f" % (newScore, oldScore))
                        print("【【【】】】newModel: %s \n" % f)
                        print(100*'_')
                        model_selection[filename] = out[filename]
                else:
                    #no score to compare
                    print("【【】】adding new model for %s \n" % filename)
                    print(100*'_')
                    model_selection[filename] = out[filename]
    return model_selection

In [3]:
t = wrapper()


Dataset size read: train 127 and test 194 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 20.691380811147901}
Lasso(alpha=20.691380811147901, copy_X=True, fit_intercept=True,
   max_iter=10000, normalize=False, positive=False, precompute=False,
   random_state=None, selection='random', tol=0.0001, warm_start=False)

【【】】adding new model for AB.pickle 

____________________________________________________________________________________________________
Dataset size read: train 192 and test 129 

# Tuning hyper-parameters 

Starting grdsurch.fit(X_train, y_train)

Best parameters set found on development set:

{'alpha': 6.1584821106602545}
Lasso(alpha=6.1584821106602545, copy_X=True, fit_intercept=True,
   max_iter=10000, normalize=False, positive=False, precompute=False,
   random_state=None, selection='random', tol=0.0001, warm_start=False)

【【】】adding new model for BC.pickle 

____________________________

In [14]:
t

{'AB.pickle': {'model': RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
             max_features='auto', max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=2,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, n_jobs=-1, oob_score=True, random_state=None,
             verbose=0, warm_start=False),
  'test_mean_y_4_comparing': 1468.9692783505157,
  'test_rmse_4_reporting': 987.57560496918211,
  'train_rmse_cv_4_picking': 613.22803481249457},
 'AC.pickle': {'model': RandomForestRegressor(bootstrap=True, criterion='mse', max_depth=None,
             max_features='auto', max_leaf_nodes=None,
             min_impurity_split=1e-07, min_samples_leaf=2,
             min_samples_split=2, min_weight_fraction_leaf=0.0,
             n_estimators=6000, n_jobs=-1, oob_score=True, random_state=None,
             verbose=0, warm_start=False),
  'test_mean_y_4_comparing': 1817.4408333333331,
  'test_rmse_4_re

In [15]:
import pickle

with open('model_selection_4C2.pickle', 'wb') as handle:
    pickle.dump(t, handle, protocol=pickle.HIGHEST_PROTOCOL)

In [17]:
for key in t.keys():
    print(key)
    print(t[key]['train_rmse_cv_4_picking'],
          t[key]['test_rmse_4_reporting'],
          t[key]['test_mean_y_4_comparing']
         )
    print(100*'_')

AB.pickle
613.228034812 987.575604969 1468.96927835
____________________________________________________________________________________________________
BC.pickle
508.487190501 1529.27220935 985.434108527
____________________________________________________________________________________________________
CD.pickle
139.115733908 2032.01164137 1801.35669291
____________________________________________________________________________________________________
AC.pickle
139.211515629 1671.60046738 1817.44083333
____________________________________________________________________________________________________
AD.pickle
85.9832652301 2228.17276351 2371.57863158
____________________________________________________________________________________________________
BD.pickle
490.776589487 1094.31527963 1277.54806202
____________________________________________________________________________________________________


# (4C2) Response when two of four modules are chosen as training dataset

'''
AB.pickle
613.228034812 987.575604969 1468.96927835
____________________________________________________________________________________________________
BC.pickle
508.487190501 1529.27220935 985.434108527
____________________________________________________________________________________________________
CD.pickle
139.115733908 2032.01164137 1801.35669291
____________________________________________________________________________________________________
AC.pickle
139.211515629 1671.60046738 1817.44083333
____________________________________________________________________________________________________
AD.pickle
85.9832652301 2228.17276351 2371.57863158
____________________________________________________________________________________________________
BD.pickle
490.776589487 1094.31527963 1277.54806202
____________________________________________________________________________________________________

'''